### PR TITLE
Added a step to validate rest api spec location before release

### PR DESCRIPTION
**Pull request by raych1**
The purpose is to gate on the REST API specifications used to release SDK are from the main branch of Azure/azure-rest-api-specs repository.

Added a step for the validation to the release pipeline.

The common script and pipeline are merged by another PR.
https://github.com/Azure/azure-sdk-tools/pull/7451

FYI - The same change has been added to dotnet release pipeline.
https://github.com/Azure/azure-sdk-for-net/pull/40783

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
I believe the script checks for both TypeSpec and Swagger source.

I can fix (if not already fixed) the case for TypeSpec.

From a scan run by Ray (email "REST API spec location validation for Azure SDK packages") there is some "commit not from main branch" in Swagger case.
Some of them are legacy and we probably won't release again (e.g. synapse, formrecognizer). But some may need fix from dev (e.g. communication, azure-storage-file-share).

Do we plan to require them do the fix before next release?
@srnagar @alzimmermsft

**Review by alzimmermsft — APPROVED** [review]

File: eng/pipelines/templates/stages/archetype-java-release.yml [hunk 1]
```diff
@@ -118,6 +118,11 @@ stages:
                     PackageName: ${{artifact.name}}
                     ServiceName: ${{parameters.ServiceDirectory}}
                     ForRelease: true
+              - template: /eng/common/pipelines/templates/steps/verify-restapi-spec-location.yml
```
Comment (srnagar) [inline eng/pipelines/templates/stages/archetype-java-release.yml:121]: Please add a comment here to specify that this script verifies both TypeSpec and swagger specification are in `main` branch.

**Review by srnagar — APPROVED** [review]
LGTM

**Review by weidongxu-microsoft — APPROVED** [review]
@raych1 

I think we can merge after you check Srikanta's comment.

File: eng/pipelines/templates/stages/archetype-java-release.yml (hunk 1 above)
Comment (raych1) [reply eng/pipelines/templates/stages/archetype-java-release.yml:122]: Added.

**Pull request merged**